internal/mqini/mqini_test.go

```diff
@@ -62,3 +64,160 @@ func TestGetQueueManager(t *testing.T) {
+	}
+	userconfig := string(iniFileBytes)
+	qmConfig, atConfig, err := PrepareConfigStanzasToWrite(userconfig)
+	if err != nil {
+		t.Errorf("Unexpected error: [%s]\n", err.Error())
+	}
+	if len(atConfig) == 0 {
+		t.Errorf("Unexpected stanza file update: mqat.ini[%s]\n", atConfig)
+	}
+	if len(qmConfig) == 0 {
+		t.Errorf("Expected stanza file not found: qm.ini\n")
+	}
+
+	scanner := bufio.NewScanner(strings.NewReader(userconfig))
+	scanner.Split(bufio.ScanLines)
+	for scanner.Scan() {
+		line := scanner.Text()
+		if !strings.Contains(qmConfig, line) {
+			t.Errorf("Expected stanza line not found in updated string. line=%s\n, Stanza:%s\n", line, qmConfig)
+			break
+		}
+	}
+}
+
+func TestIniFile2Update(t *testing.T) {
+
+	iniFileBytes, err := ioutil.ReadFile("sample2qm.ini")
+	if err != nil {
+		t.Errorf("Unexpected error: [%s]\n", err.Error())
+	}
+	userconfig := string(iniFileBytes)
+	qmConfig, atConfig, err := PrepareConfigStanzasToWrite(userconfig)
+	if err != nil {
+		t.Errorf("Unexpected error: [%s]\n", err.Error())
+	}
+	if len(atConfig) == 0 {
+		t.Errorf("Unexpected stanza file update: mqat.ini[%s]\n", atConfig)
+	}
+	if len(qmConfig) == 0 {
+		t.Errorf("Expected stanza file not found: qm.ini\n")
+	}
+
+	scanner := bufio.NewScanner(strings.NewReader(userconfig))
+	scanner.Split(bufio.ScanLines)
+	for scanner.Scan() {
+		line := scanner.Text()
+		if !strings.Contains(qmConfig, line) {
+			t.Errorf("Expected stanza line not found in updated string. line=%s\n, Stanza:%s\n", line, qmConfig)
+			break
+		}
+	}
+}
+
+func TestIniFile3Update(t *testing.T) {
+
+	iniFileBytes, err := ioutil.ReadFile("sample3qm.ini")
+	if err != nil {
+		t.Errorf("Unexpected error: [%s]\n", err.Error())
+	}
+	userconfig := string(iniFileBytes)
+	qmConfig, atConfig, err := PrepareConfigStanzasToWrite(userconfig)
+	if err != nil {
+		t.Errorf("Unexpected error: [%s]\n", err.Error())
+	}
+	if len(atConfig) == 0 {
+		t.Errorf("Expected stanza file not found: mqat.ini\n")
+	}
+	if len(qmConfig) == 0 {
+		t.Errorf("Expected stanza file not found: qm.ini\n")
+	}
+
+	scanner := bufio.NewScanner(strings.NewReader(userconfig))
+	scanner.Split(bufio.ScanLines)
+	for scanner.Scan() {
+		line := scanner.Text()
+		if !strings.Contains(atConfig, line) {
+			t.Errorf("Expected stanza line not found in updated string. line=%s\n, Stanza:%s\n", line, qmConfig)
+			break
+		}
+	}
+}
+
+func TestIniFile4Update(t *testing.T) {
+
+	iniFileBytes, err := ioutil.ReadFile("sample4qm.ini")
+	if err != nil {
+		t.Errorf("Unexpected error: [%s]\n", err.Error())
+	}
+	userconfig := string(iniFileBytes)
+	qmConfig, atConfig, err := PrepareConfigStanzasToWrite(userconfig)
+	if err != nil {
+		t.Errorf("Unexpected error: [%s]\n", err.Error())
+	}
+	if len(qmConfig) == 0 {
+		t.Errorf("Unexpected stanza file update: qm.ini[%s]\n", atConfig)
+	}
+	if len(atConfig) == 0 {
+		t.Errorf("Expected stanza file not found: mqat.ini\n")
+	}
+
+	scanner := bufio.NewScanner(strings.NewReader(userconfig))
```

Comment:
Here, it looks like you're checking that all the lines in the input file are present in one of the output strings, but it's not checking that it's in the _right_ output string.

Reply:
Do you think that would be useful, or do you think this is covered by your other test which ensures the stanzas are directed to the right file?

Reply:
Thank you. Yes, I have modified the test to check right strings in right files and I have also removed redundant test.